src/sketch.js

```diff
@@ -97,6 +97,6 @@ function controls() {
   if (keyIsDown(88)) {
     zoom -= zoomSpeed
   }
-  zoom = constrain(zoom, 0.05, 1)
+  zoom = constrain(zoom, observableMin / min(UNIVERSE.WIDTH, UNIVERSE.HEIGHT), 1)
```

Comment:
```suggestion
zoom = constrain(zoom, observableMin / min(UNIVERSE.WIDTH, UNIVERSE.HEIGHT), 1)
```
`observableMin / min(UNIVERSE.WIDTH, UNIVERSE.HEIGHT)` should be its own variable, it would be more readable.